### ISSUE TITLE
Vagrant reload does not check WinRM port

### ISSUE THREAD

**Issue by tendervittles** [WinRb/vagrant-windows#124]
If have seen this behavior on both CentOS 6.4 and OS X 10.9 hosts.

I am running Vagrant 1.3.4.

Please let me know if this should be filed against Vagrant.

In my Vagrantfile I specify the WinRM port and a forward in a multi-machine environment:

``` ruby
    win.vm.network :forwarded_port, guest: 5985, host: 10000
    win.winrm.port = 10000
```

This works. My machine comes up reliably.

``` text
$ vagrant up win
Bringing machine 'win' up with 'virtualbox' provider...
[win] Clearing any previously set forwarded ports...
[win] Creating shared folders metadata...
[win] Clearing any previously set network interfaces...
[win] Preparing network interfaces based on configuration...
[win] Forwarding ports...
[win] -- 22 => 2222 (adapter 1)
[win] -- 5985 => 10000 (adapter 1)
[win] Booting VM...
[win] Waiting for machine to boot. This may take a few minutes...
[win] Machine booted and ready!
[win] Mounting shared folders...
[win] -- /vagrant
$
```

Now I edit the Vagrantfile and use port 10001, then reload

``` ruby
    win.vm.network :forwarded_port, guest: 5985, host: 10001
    win.winrm.port = 10001
```

``` text
$ vagrant reload win
[win] Attempting graceful shutdown of VM...
[win] Clearing any previously set forwarded ports...
[win] Creating shared folders metadata...
[win] Clearing any previously set network interfaces...
[win] Preparing network interfaces based on configuration...
[win] Forwarding ports...
[win] -- 22 => 2222 (adapter 1)
[win] -- 5985 => 10001 (adapter 1)
[win] Booting VM...
[win] Waiting for machine to boot. This may take a few minutes...
Timed out while waiting for the machine to boot. This means that
Vagrant was unable to communicate with the guest machine within
the configured ("config.vm.boot_timeout" value) time period. This can
mean a number of things.

If you're using a custom box, make sure that networking is properly
working and you're able to connect to the machine. It is a common
problem that networking isn't setup properly in these boxes.
Verify that authentication configurations are also setup properly,
as well.

If the box appears to be booting properly, you may want to increase
the timeout ("config.vm.boot_timeout") value.
$
```

If I vagrant halt and vagrant up then everything is fine.

``` text
$ vagrant halt win
[win] Attempting graceful shutdown of VM...
$ vagrant up win
Bringing machine 'win' up with 'virtualbox' provider...
[win] Clearing any previously set forwarded ports...
[win] Creating shared folders metadata...
[win] Clearing any previously set network interfaces...
[win] Preparing network interfaces based on configuration...
[win] Forwarding ports...
[win] -- 22 => 2222 (adapter 1)
[win] -- 5985 => 10001 (adapter 1)
[win] Booting VM...
[win] Waiting for machine to boot. This may take a few minutes...
[win] Machine booted and ready!
[win] Mounting shared folders...
[win] -- /vagrant
$
```

Thanks!

**Comment by sneal** [comment]
Do you know if core Vagrant detects an SSH port change between up and reload?

**Comment by sneal** [comment]
Can you try the latest 1.5.1 gem? I believe this was fixed by #157. The only annoying thing is that it takes a couple of minutes while it times out. Here's the output of my run after changing the loaded VM's winrm port:

```
$ vagrant reload
[default] Attempting graceful shutdown of VM...
[default] Clearing any previously set forwarded ports...
[Berkshelf] Skipping Berkshelf with --no-provision
[default] Clearing any previously set network interfaces...
[default] Preparing network interfaces based on configuration...
[default] Forwarding ports...
[default] -- 22 => 2222 (adapter 1)
[default] -- 5985 => 10001 (adapter 1)
[default] Booting VM...
[default] Waiting for machine to boot. This may take a few minutes...
[default] Machine booted and ready!
[default] Mounting shared folders...
[default] -- /vagrant
[default] -- /tmp/vagrant-puppet-1/manifests
[default] VM already provisioned. Run `vagrant provision` or use `--provision` to force it
```
